### PR TITLE
added the ability to use XLSX file format as a document source

### PR DESCRIPTION
**Pull request by Allaye**
This pull request adds the ability to use XLSX files as a document source. This feature allows users to import data from XLSX files into the application, providing more flexibility and convenience. The implementation does not change how users interact with the code.

### CODE REVIEW

**Comment by YaJannik** [comment]
I've been reviewing this PR with a focus on adding markdown support, and I have a few concerns that I'd like to address:

- Adding the .gitignore is a great addition, but it appears to remove the _constitution.pdf_ file. I believe having some demo data would be beneficial for testing purposes.
- While you've implemented the conversion of _xlxs files_, it seems that the file_path parameter is missing in the _ingest.py_ file.
- I noticed that in the _ingest.py_ file, the default device type has been changed to cpu instead of gpu. It would be worth discussing if this change is intentional or if it needs further consideration.
- Lastly, to wrap up this PR, it would be helpful to include the instructions for ingesting the new file format in the readme.md file under the section titled _Instructions for Ingesting Your Own Dataset_.

Please let me know your thoughts on these comments.

**Comment by Allaye** [comment]
@YaJannik i think you are right i will fix this soon

**Comment by Allaye** [comment]
> I've been reviewing this PR with a focus on adding markdown support, and I have a few concerns that I'd like to address:
> 
> * Adding the .gitignore is a great addition, but it appears to remove the _constitution.pdf_ file. I believe having some demo data would be beneficial for testing purposes.   
> * While you've implemented the conversion of _xlxs files_, it seems that the file_path parameter is missing in the _ingest.py_ file.
> * I noticed that in the _ingest.py_ file, the default device type has been changed to cpu instead of gpu. It would be worth discussing if this change is intentional or if it needs further consideration.
> * Lastly, to wrap up this PR, it would be helpful to include the instructions for ingesting the new file format in the readme.md file under the section titled _Instructions for Ingesting Your Own Dataset_.
> 
> Please let me know your thoughts on these comments.


@YaJannik take a look at the PR now and confirm if the raised issues have been taking care of.

File: ingest.py [hunk 1]
```diff
@@ -24,19 +24,28 @@ def load_single_document(file_path: str) -> Document:
...
-    return [load_single_document(f"{source_dir}/{file_path}") for file_path in all_files if file_path[-4:] in ['.txt', '.pdf', '.csv'] ]
+    docs = []
+    for file_path in all_files:
+        if file_path[-4:] == 'xlsx':
+            for doc in xlxs_to_csv(f"{source_dir}/{file_path}"):
+                docs.append(load_single_document(doc))
+        elif file_path[-4:] in ['.txt', '.pdf', '.csv']:
+            docs.append(load_single_document(f"{source_dir}/{file_path}"))
+    return docs
+    # return [load_single_document(f"{source_dir}/{file_path}") for file_path in all_files if
```
Comment (Truewarlock) [inline ingest.py:35]: Is commented code still necessary?

Comment (Allaye) [reply ingest.py:35]: There is nothing wrong with it

**Comment by PromtEngineer** [comment]
@Allaye I looked at the PR and looks good to me. @YaJannik raised some good points so will wait for him to review it once again and then I can merge the PR. Thanks for the help!

**Comment by YaJannik** [comment]
Sorry for the delay. Everything is fixed and I‘m happy that i could help. Closed for me.

**Pull request merged**